packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -715,7 +750,19 @@ export const MenuBarMixin = (superClass) =>
      */
     _setFocused(focused) {
       if (focused) {
-        const target = this.querySelector('[tabindex="0"]');
+        if (this.tabNavigation) {
```

Comment:
Using `setFocused` for changing the opened sub-menu on <kbd>Tab</kbd> / <kbd>Shift Tab</kbd> doesn't cover the case when this button is pressed on the sub-menu item. I believe we should use `__onContextMenuKeydown` instead - that's where <kbd>Arrow Left</kbd> and <kbd>Arrow Right</kbd> are handled.

For example, here's how it should work on <kbd>Shift Tab</kbd> on the sub-menu item:

https://github.com/vaadin/web-components/assets/10589913/550e28b7-36e9-4525-80b0-67fab553c609

Reply:
I've changed the code of `__onContextMenuKeydown` to manage manually the submenu behavior.

It's a bit odd because the `vaadin-contextmenu-items-mixin.js` is calling close-all-menus of Tab:
```
if (key === 'Tab') {
          // Close all menus
          this.dispatchEvent(new CustomEvent('close-all-menus'));
        }
````

And `close-all-menus` is calling `close` that generates a call `_setFocused` (to focus on the root item that has been closed. Because of this, you're losing the context (the submenu is closed).